### PR TITLE
Workaround for enum string deserialization using snake case

### LINKED ISSUE #25501: Provide workaround for enum string deserialization using snake case
See https://github.com/dotnet/runtime/issues/31619

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7bee4ecb-7464-b321-937a-e6d4586556eb
* Version Independent ID: 27c18218-d28f-a524-b7e1-8262005b7026
* Content: [How to customize property names and values with System.Text.Json](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-customize-properties)
* Content Source: [docs/standard/serialization/system-text-json-customize-properties.md](https://github.com/dotnet/docs/blob/main/docs/standard/serialization/system-text-json-customize-properties.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @tdykstra
* Microsoft Alias: **tdykstra**

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #25501 

File Type | File Name | Published Url
-- | -- | --
Content | docs/standard/serialization/system-text-json-converters-how-to.md | https://review.docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-converters-how-to?branch=pr-en-us-25500
Content | docs/standard/serialization/system-text-json-customize-properties.md | https://review.docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-customize-properties?branch=pr-en-us-25500

### CODE REVIEW

File: docs/standard/serialization/snippets/system-text-json-converters-how-to/csharp/EnumConverter.cs
Comment (BillWagner) [inline docs/standard/serialization/snippets/system-text-json-converters-how-to/csharp/EnumConverter.cs:190]: Do you want to hold up this PR for those changes, or make those enhancements in another PR?

**Review by BillWagner — APPROVED** [review]
This LGTM @tdykstra 

Once @layomia approves, you can :shipit:

File: docs/standard/serialization/snippets/system-text-json-converters-how-to/csharp/EnumConverter.cs
Comment (tdykstra) [reply docs/standard/serialization/snippets/system-text-json-converters-how-to/csharp/EnumConverter.cs:190]: I'm changing the PR to link to the issue that provides the code.

**Pull request merged**
